src/test_utils.rs

```diff
@@ -660,40 +650,42 @@ pub fn bob_keypair() -> KeyPair {
     }
 }
 
-pub struct EvTracker(Receiver<EventType>);
+/// Utility to help wait for and retrieve events.
+///
+/// This buffers the events in order they are emitted.  This allows consuming events in
+/// order while looking for the right events using the provided methods.
+///
+/// It only exposes the [`EventType`] since it can only be attached to a single
+/// [`TestContext`] and therefore the context is already known.
```

Comment:
Where does it expose an `EventType`?

And why does the fact that it can only be attached to a single `TestContext` imply that the context is already known?

Reply:
1. All methods return `EventType`.
2. You access it using e.g. `alice.evtracker`, where `alice` is the `TestContext`.

I've improved the doc comment to make this clearer.